### DISCUSSION TITLE
Instances predicted on 0 frames for pupil tracking

### DISCUSSION THREAD

**Discussion by audreydrotos** [talmolab/sleap#2692] [Help!] [answered]
Hello! I just started using SLEAP for pupil size estimation in videos of headfixed mice. I am following the tutorial and having an issue where training will run and inference also says completed, but it will say "instances were predicted on 0 frames (40 frames with no instances found)." I have 40 manually labeled frames and have tried predicting on those frames, current frame, etc. without luck. I have pasted my parameters below I am using for traning. I have also tried adjusting the input scaling and max stride after reading a few discussions here, but that doesn't seem to be helping. If anyone has any ideas that the issue would be, that would be super helpful! Thank you!

<img width="993" height="932" alt="image" src="https://github.com/user-attachments/assets/9cb2f02d-fa62-4f9c-a5eb-31b76d247085" />

<img width="993" height="932" alt="image" src="https://github.com/user-attachments/assets/d3549644-bcfc-4b3b-8f79-1fddfaed7d50" />

**Comment by gitttt-1234** [comment, marked as answer]
Hi @audreydrotos !

Could you try the following and see if that improves the performance?

- Increase number of epochs: 5 is far too few for the model to learn. Set it to **100 / 200** (You already have the stop training on plateau ON, so the training would automatically be stopped once the model stops improving)
- Try setting rotation to **15** (Since this is a head-fixed pupil prep, the eye orientation barely changes. ±180°
is asking the model to learn the pupil from every possible rotation, which might not even be possible in a real setting)
- Try increasing filters to 32. (this increases the model size, and might slow down the model training a bit)
- Do you need input scaling to be 0.25? Lower scale loses fine detail for precise localization (However, this totally depends on your application)

Let us know if you have any questions!

Thanks,

Divya

**Reply by audreydrotos** [reply]
Hi, thank you for the help! I adjusted these parameters and set epochs to 200, rotation to 15, filters to 32, and input scaling back to 1. I had initially set the epochs lower for testing because it was taking a while to run (currently it is on epoch 38 with these settings and has been running for 968 minutes.) I have an NVIDIA GeForce RTX 3060 graphics card and installed the PyTorch version that should use it, and looking at the task manager the GPU is currently maxed out so I think it is being utilized to train.

**Reply by gitttt-1234** [reply]
Hi @audreydrotos ! 

That's great! If you want to speed-up training, you can try input scaling set to 0.5. However, this might lead to very little precision loss since you would be processing image at half-resolution, 0.25 would've been too small, so 0.5 could be a sweet spot too.

Thanks,

Divya

**Reply by audreydrotos** [reply]
Hi Divya, thanks so much! I will try that as well. Appreciate the help!!

**Reply by audreydrotos** [reply]
Hi Divya, I'm actually still having the same issue as before with the adjusted parameters (instances were predicted on 0 frames). Here is the output I'm getting. I tried using 200 epochs as suggested. I'm looking through the discussion pages here and I think I've adjusted everything appropriately for people who have had similar issues but seems like I am still missing something. Thank you again for the help! 

<img width="602" height="532" alt="image" src="https://github.com/user-attachments/assets/3e8363fa-4a04-41d0-b43d-42d839fe556c" />

**Reply by gitttt-1234** [reply]
Hi @audreydrotos !

Did you try increasing the filters to 32? How many labeled frames are you training with?

Thanks,

Divya

**Reply by audreydrotos** [reply]
Hi, thanks for the help! I increased my labeled frames to 300 and that solved the issue--I was using 200 before, maybe this was not enough training data. I did have another follow up question: my model is working well and I would like to export a video with the poses overlaid, which I see is an option in the docs in the predict menu (predict -> export video with visual annotations). In my predict menu, I'm not seeing this option though. I'm using v1.6.1. Is there a setting needed to toggle this, or something else that might be causing it to disappear? Thank you! 
<img width="415" height="221" alt="image" src="https://github.com/user-attachments/assets/734f075b-7bed-46db-82b1-b6bfee651d58" />

**Reply by gitttt-1234** [reply]
Hi @audreydrotos !

Apologies for this, To export a video with instances overlaid, go to **View** -> **Render Video clip with instances**. We will fix the docs in the next PR, Thanks for reporting!
